enumerable.rb

```diff
@@ -0,0 +1,145 @@
+module MyEnumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    i = 0
+    while i < array.length
+      yield(array[i])
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    if is_a?(Range)
+      array = to_array
+      array.length.my_each do |i|
+        yield(array[i], i)
+      end
+    else
+      i = 0
+      while i < array.length
+        yield(array[i], i)
+        i += 1
+      end
+    end
+    array
+  end
```

Comment:
* [ ] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_each_with_index calls the given block once for each element in self
  ```ruby
    array = [1,2,5,8,7,98,4]
    block = proc { |num, idx| my_each_output += "Num: #{num}, idx: #{idx}\n" }
    p array.my_each_with_index(&block) # returns this error
    # NoMethodError:
       undefined method `length' for nil:NilClass
     # ./enumerable.rb:24:in `my_each_with_index'
  ```